lib/es6/closer.js

```diff
@@ -0,0 +1,94 @@
+import assrt from 'assert';
+import octonode from 'octonode';
+import Q from 'q';
+import _ from 'lodash';
+import path from 'path';
+import fs from 'fs';
+import { getLogger } from './logger';
+import { mapToObject } from './utils';
+
+let closerSingleton;
+
+class IssueCloser {
+  constructor (repoConfig:Object):void {
+    this.bot = repoConfig.bot;
+    this.repos = repoConfig.repos;
+    this.logger = getLogger();
+  }
+
+  repoStr (repo:Object):string {
+    return `${repo.user}/${repo.repo}`
+  }
+
+  async closeIssue (data) {
+    let [ourRepo, theirUser, theirRepo] = this.repoForIssue(data);
+    if (!this.shouldCloseIssue(ourRepo, data)) {
+      this.logger.info("No close request. Doing nothing.");
+      return;
+    }
+    this.logger.info(`@${theirUser} wants to close #${data.issue.url}`);
+
+    if (this.canCloseIssue(ourRepo, theirUser)) {
+      this.logger.info('Permission granted.');
+      await this.performClosure(this.repoStr(ourRepo), data.issue.number);
+    } else {
+      this.logger.info('Permission denied.');
+    }
+  }
+
+  repoForIssue (issue:Object):Array {
+    let user:string = issue.repository.owner.login;
+    let repo:string = issue.repository.name;
+    for (let ourRepo of this.repos) {
+      if (user === ourRepo.user && repo === ourRepo.repo) {
+        return [ourRepo, user, repo];
+      }
+    }
+    return [null, user, repo];
+  }
+
+  shouldCloseIssue (ourRepo:Object, data:Object):boolean {
+    let needle = `@${this.bot} please close.`;
+    return data.comment.body.indexOf(needle) > -1;
+  }
+
+  canCloseIssue (ourRepo:Object, theirUser:string):boolean {
+    return ourRepo.triagers.indexOf(theirUser) > -1;
+  }
+
+  async performClosure (repoStr:string, issueNum:number):void {
+    this.logger.info("Closing issue on github");
+    let client = octonode.client(process.env.TRIAGER_TOKEN);
+    let issue = client.issue(repoStr, issueNum);
+    let updateObj = {};
+    updateObj.state = 'closed';
+    let [updated, res] = await Q.ninvoke(issue, "update", updateObj);
+    if (res.status !== "200 OK") {
+      this.logger.error(res);
+      throw new Error("Github said things didn't go well");
+    }
+    this.logger.info("Issue closed!");
+  }
+
+  validateIssueCommentFormat (data:Object):void {
```

Comment:
The comments suggest this method was originally put in place to differentiate between webhooks (issues vs anything else).

https://github.com/scottdixon/triager/blob/issue-closer/lib/es6/routes.js is now more strict - _only issues_ will get through to assigner and _only issue comments_ will get through to closer.

If it's still important to validate the data we get from GitHub, we can abstract the 2x validation methods into their own class. Otherwise we can scrap these.